Add bindings for rlgl.h

Willenbrink opened the pull request:
Hey, this PR adds bindings for the rlgl.h and ports one example (instanced rendering).

The bindings themselves were mostly made using regexes so it isn't as tidy as the rest of the bindings. I don't really understand how this repository is structured so I've added the relevant files at the most sensible locations. The rlgl bindings reside in their own Rl module and the rl_ prefix of the functions has been removed. Unfortunately rl_begin and rl_end include the prefix as begin and end are keywords in OCaml. Perhaps they could be renamed to start/stop, remain as is or we keep the rl_ prefix everywhere.

I had these unpolished bindings lying around for some time and ported the instanced rendering example when it was still reliant on rlgl. As of now, draw_mesh_instanced is included in the raylib.h so this example is sort of superfluous. It still shows some of the uglier sides of handling carrays and shaders so I guess it's still nice to have.

PS: This PR depends on #20. Is there a nicer way to tell github about this than duplicating the commits from the other PR?

tjammer commented on src/raylib/raylib.mli:
```diff
@@ -3011,3 +3011,5 @@ val set_audio_stream_pitch : AudioStream.t -> float -> unit
 val set_audio_stream_buffer_size_default : int -> unit
 (** [set_audio_stream_buffer_size_default size] Default size for new audio streams*)
+
+module Rl = Rlgl
```
I don't think the abbreviation makes much sense here. `Rlgl` stands for "Raylib GL", no?
I think `Rlgl` is fine here, since it's named like that throughout the raylib C library. Otherwise, I'd prefer `Gl`, so we can access it like `Raylib.Gl`, which fits `rlgl`.

Willenbrink replied:
Sure, I chose that because of the rl_ prefix but that doesn't really make that much sense in retrospect.

tjammer commented:
Thanks for this and sorry for taking a long time to respond!

I think the structuring is fine as it is. At some point I want to separate `Math` and `Rlgl` into submodules, so we have to link them only if we need it, but that's a different story. Right now it makes to leave them in the main `Raylib` module.

As for `begin` and `end`, I think it's best to keep the prefix. Maybe we could make it a suffix, so users who search for `begin` will find `begin_rl` and don't have to search until they reached `r`. I'm not really sure which version is best.

tjammer commented on src/raylib/rlgl.mli:
```diff
@@ -0,0 +1,300 @@
+val load_render_batch :
+  int ->
+  int ->
+  Raylib_generated_types__.Rlgl_generated_types.RlRenderBatch.t Ctypes.structure
```
Instead of having these long generated types, I would create separate modules for each rlgl type, like the main raylib does for eg `Vector2`.
We still can merge this and polish it later

Willenbrink commented:
Thanks for the feedback and sorry for taking so long to address the issues!

I've made some minor changes to the structure (Rlgl_generated_types is now separate from Raylib_generated_types and no longer a submodule of it), renamed types.ml to raylib_types.ml and addressed the comments.

I've left `rl_begin`/`rl_end` with their prefix as I believe that is the most straightforward solution.

tjammer commented:
As far as I'm concerned the PR is ready to be merged, thanks!

One more thing: github can't link the commits to your account, thus you don't show up in the contributors list. Is this on purpose? If it's not, feel free to re-push the commits with a fixed author. Although I'm not sure what feature github uses, e-mail maybe.
